component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -632,16 +632,29 @@ private void checkStatusCode(HttpServletResponse response, AuthenticationContext
 
         boolean isRetryEnabled = SMSOTPUtils.isRetryEnabled(context);
         String loginPage = getLoginPage(context);
+        AuthenticatedUser authenticatedUser =
+                (AuthenticatedUser) context.getProperty(SMSOTPConstants.AUTHENTICATED_USER);
         String url = getURL(loginPage, queryParams);
         if (StringUtils.isNotEmpty(getScreenValue(context))) {
             url = url + SMSOTPConstants.SCREEN_VALUE + getScreenValue(context);
         }
         try {
-            if (Boolean.parseBoolean(String.valueOf(context.getProperty(SMSOTPConstants.ACCOUNT_LOCKED))) &&
-                    SMSOTPUtils.isShowAuthFailureReason(context)) {
-                response.sendRedirect(url + SMSOTPConstants.RESEND_CODE
-                        + SMSOTPUtils.isEnableResendCode(context) + SMSOTPConstants.ERROR_MESSAGE + SMSOTPConstants
-                        .ACCOUNT_LOCKED_ERROR);
+            if (SMSOTPUtils.isLocalUser(context) && SMSOTPUtils.isAccountLocked(authenticatedUser)) {
+                boolean showAuthFailureReason = SMSOTPUtils.isShowAuthFailureReason(context);
+                String retryParam;
+                if (showAuthFailureReason) {
+                    String unlockTime = getUnlockTime(authenticatedUser);
```

Comment:
Can this return null? If so shall we add a debug log with more context to identify that?

Reply:
Yes. `null` can be returned if no value is configured for `http://wso2.org/claims/identity/unlockTime`. This can happen if the admin locks the account.

Reply:
fixed in f3b362c